### DISCUSSION TITLE
Internationalized Routing: Sub-Path redirects

### DISCUSSION THREAD

**Discussion by GioLogist** [vercel/next.js#22251] [Help]
In [Domain Routing](https://nextjs.org/docs/advanced-features/i18n-routing#domain-routing), we are able to redirect multiple locales to a single domain, like so:

```
      {
        domain: 'example.nl',
        defaultLocale: 'nl-NL',
        // specify other locales that should be redirected
        // to this domain
        locales: ['nl-BE'],
      },
```

Is there an equivalent of doing this, but for sub-paths (not domains)? For example, if I wanted both `en-GB-oxendict` and `en-GB` to both use the sub-path `/uk`, what is the recommended approach?

**Comment by ArenoRobert** [comment]
@GioLogist did you find a solution for this?

**Reply by GioLogist** [reply]
Nope. As a workaround, I ended up keeping a `uk` locale only. Then, I added a component to the header that has a flag for UK/US. Upon a user selecting that flag, we set the `NEXT_LOCALE` cookie to `uk` or `us`. This allows us to be certain of a user's locale, regardless of the `Accept-Language` header, which is what Next.JS uses for automatic detection. 

Another method we tried (but didn't work) was adding the redirects to the NextJS config. I'm assuming the redirect method didn't work because of [this](https://github.com/vercel/next.js/issues/20165). Although it says "fixed" in 10.0.4, we're running 10.0.6 and still unable to get redirects working w/a localized route.

```js
  async redirects() {
    return [
      {
        source: '/en-GB/:route',
        destination: '/uk/:route',
        permanent: false,
      },
    ]
  }
```
